src/sqs/push.ts

```diff
@@ -39,3 +41,60 @@ export const pushQueueMessageHandler : MessageHandler<PushQueueMessagePayload> =
 	await processPush(github, payload, wrapLogger(context.log));
 }
 
+
+const NON_ERRONEOUS_STATUSES = [401, 404, 403];
+const RATE_LIMITING_DELAY_BUFFER = 10;
+const EXPONENTIAL_BACKOFF_BASE = 60;
+export const pushQueueErrorHandler : ErrorHandler<PushQueueMessagePayload> = async (error: JiraClientError | Octokit.HookError | RateLimitingError | Error,
+	context: Context<PushQueueMessagePayload>) : Promise<ErrorHandlingResult> => {
+
+	//Handle non-failure cases
+	const maybeResult = maybeHandleNonFailureCase(error, context);
+	if(maybeResult) {
+		return maybeResult;
+	}
+
+	//Handle failure cases
+	const errorHandlingResult = handleFailureCase(error, context);
+
+	if(!errorHandlingResult.retryable || context.lastAttempt ) {
+		//Emit metrics if the failure is not retryable
+		context.log.error({error}, "Webhook push processing failed and won't be retried anymore");
+		emitWebhookFailedMetrics("push")
+	}
+
+	return errorHandlingResult;
+}
+
+function maybeHandleNonFailureCase(error: Error, context: Context<PushQueueMessagePayload>): ErrorHandlingResult | undefined {
+	if(error instanceof JiraClientError &&
+		error.status &&
+		NON_ERRONEOUS_STATUSES.includes(error.status)) {
+		context.log.warn(`Received ${error.status} from Jira. Unretryable. Discarding the message`);
+		return {retryable: false}
+	}
+
+	//If error is Octokit.HookError, then we need to check the response status
+	//Unfortunately we can't check if error is instance of Octokit.HookError because it is not a calss, so we'll just rely on status
+	//TODO Add error handling for the new GitHub client when it will be done
+	const maybeErrorWithStatus : any = error;
```

Comment:
nit: Could it be `const { status } = error;` instead?

Reply:
That was done to avoid typescript errors. If I do `const { status } = error;` It will say that the error doesnt' have "status" field. However its subclasses can. So I had to do this to avoid ignoring typescript&linter errors